lib/src/Hodor.cc

```diff
@@ -0,0 +1,186 @@
+#include <drogon/plugins/Hodor.h>
+#include <drogon/plugins/RealIpResolver.h>
+
+using namespace drogon::plugin;
+void Hodor::initAndStart(const Json::Value &config)
+{
+    if (config.isMember("urls") && config["urls"].isArray())
+    {
+        std::string regexString;
+        for (auto &str : config["urls"])
+        {
+            assert(str.isString());
+            regexString.append("(").append(str.asString()).append(")|");
+        }
+        if (!regexString.empty())
+        {
+            regexString.resize(regexString.length() - 1);
+            regex_ = std::regex(regexString);
+            regexFlag_ = true;
+        }
+    }
+    algorithm_ = stringToRateLimiterType(
+        config.get("algorithm", "token_bucket").asString());
+    timeUnit_ = std::chrono::seconds(config.get("time_unit", 60).asUInt());
+    capacity_ = config.get("capacity", 0).asUInt();
+    multiThreads_ = config.get("multi_threads", true).asBool();
+    if (capacity_ > 0)
+    {
+        if (multiThreads_)
+        {
+            globalLimiterPtr_ = std::make_shared<SafeRateLimiter>(
+                RateLimiter::newRateLimiter(algorithm_, capacity_, timeUnit_));
+        }
+        else
+        {
+            globalLimiterPtr_ =
+                RateLimiter::newRateLimiter(algorithm_, capacity_, timeUnit_);
+        }
+    }
+    ipCapacity_ = config.get("ip_capacity", 0).asUInt();
+    if (ipCapacity_ > 0)
+    {
+        ipLimiterMapPtr_ =
+            std::make_unique<CacheMap<std::string, RateLimiterPtr>>(
+                drogon::app().getLoop(),
+                timeUnit_.count() / 60 < 1 ? 1 : timeUnit_.count() / 60,
+                2,
+                100);
+    }
+
+    userCapacity_ = config.get("user_capacity", 0).asUInt();
+    if (userCapacity_ > 0)
+    {
+        userLimiterMapPtr_ =
+            std::make_unique<CacheMap<std::string, RateLimiterPtr>>(
+                drogon::app().getLoop(),
+                timeUnit_.count() / 60 < 1 ? 1 : timeUnit_.count() / 60,
+                2,
+                100);
+    }
+
+    useRealIpResolver_ = config.get("use_real_ip_resolver", false).asBool();
+    rejectResponse_ = HttpResponse::newHttpResponse();
```

Comment:
May be provide a  `void setRejectResponseFactory(...)` api ?
Use a static HttpResponsePtr as default.